test/smoke/omp_is_present/omp_is_present.c

```diff
@@ -0,0 +1,50 @@
+#include <stdio.h>
+#include <stdbool.h>
+#include <omp.h>
+
+#define N 1000
+#define M 123456
+
+struct TT
+{
+  int *a;
+};
+
+#pragma omp declare target
+int a_gbl = 0;
+#pragma omp end declare target
+
+int main() {
+  bool correct = false;
+  int n = N;
+  int m = M;
+  struct TT p;
+  int a = 3;
+  int *b = (int *) malloc(n*sizeof(int));
+  p.a = (int *) malloc(m*sizeof(int));
+
+  
+  #pragma omp target data map(a, b[:n], p.a[:m])
+  {
+    correct = omp_target_is_present(&a, 0);
+    correct = correct && omp_target_is_present(b, 0);
+    correct = correct && omp_target_is_present(&p, 0);
+    correct = correct && omp_target_is_present(p.a, 0);
+  }
+
```

Comment:
Perhaps add a few calls to omp_target_is_present outside the mapped environments where it should return false.

Reply:
I am also working on a tester.  We need a fail inside data map of an adjacent memory reference. I suggest mapping b[:n-1] and then check &b[n-1] for present.   Then printf address and value of b[n-2] and b[n-1] inside a target region.    We need a scenario where the array crosses a page boundary and only the front page is mapped. Reference to the 2nd page will cause a fine-grain memory page fault.